pycloudlib/qemu/cloud.py

```diff
@@ -0,0 +1,661 @@
+            download_dir.mkdir(parents=True, exist_ok=False)
+        except FileExistsError:
+            img_files = list(download_dir.glob("*.img"))
+            if img_files:
+                self._log.debug("Image already exists, skipping download")
+                return str(img_files[0].absolute())
+        self._download_file(img_url, download_dir / img_name)
+        self._download_file(kernel_url, download_dir / kernel_name)
+        return str(Path(download_dir, img_name).absolute())
+
+    def released_image(self, release, **kwargs):
+        """ID of the latest released image for a particular release.
+
+        If an image for this series from any date has been downloaded,
+        it will be used.
+
+        Args:
+            release: The release to look for
+
+        Returns:
+            A single string with the latest released image ID for the
+            specified release.
+
+        """
+        base_url = (
+            f"https://cloud-images.ubuntu.com/releases/{release}/release"
+        )
+        release_number = UBUNTU_RELEASE_VERSION_MAP[release]
+        return self._get_latest_image(
+            base_url=base_url,
+            release=release_number,
+            img_name=f"ubuntu-{release_number}-server-cloudimg-amd64.img",
+        )
+
+    def daily_image(self, release: str, **kwargs):
+        """ID of the latest daily image for a particular release.
+
+        Download the latest daily image unless it already exists.
+
+        Args:
+            release: The release to look for
+
+        Returns:
+            A single string with the latest daily image ID for the
+            specified release.
+
+        """
+        return self._get_latest_image(
+            base_url=f"https://cloud-images.ubuntu.com/{release}/current",
+            release=release,
+            img_name=f"{release}-server-cloudimg-amd64.img",
+        )
+
+    def image_serial(self, image_id):
+        """Find the image serial of the latest daily image for a release.
+
+        Args:
+            image_id: string, Ubuntu image id
+
+        Returns:
+            string, serial of latest image
+
+        """
+        raise NotImplementedError
+
+    def get_instance(self, instance_id, **kwargs) -> QemuInstance:
+        """Get an instance by id.
+
+        Args:
+            instance_id: The id for this instance. For QEMU this takes the
+                form of <instance_path>::<ssh_port>::<telnet_port>
+
+        Returns:
+            An instance object to use to manipulate the instance further.
+
+        """
+        return QemuInstance(
+            key_pair=self.key_pair,
+            instance_id=instance_id,
+            handle=None,
+            username=kwargs.get("username"),
+        )
+
+    def _create_seed_iso(
+        self,
+        instance_dir: Path,
+        user_data: Optional[str],
+        meta_data: Optional[str],
+        vendor_data: Optional[str] = None,
+        network_data: Optional[str] = None,
```

Comment:
Shouldn't we call this network_config because that is the name of the configuration that cloud-init is consuming?